docs/usage.md

```diff
@@ -0,0 +1,212 @@
+# Usage
+
+To compare two versions of a program for semantic differences using DiffKemp
+use the following commands:
+
+1. [Snapshot generation command](#1-snapshot-generation-commands)  
+   a) [For `make`-based projects](#a-build-snapshot-generation-of-make-based-projects)  
+   b) [For the Linux kernel](#b-build-kernel-snapshot-generation-from-the-linux-kernel)  
+   c) [For a single LLVM IR file](#c-llvm-to-snapshot-snapshot-generation-from-a-single-llvm-ir-file)
+2. [Semantic comparison command](#2-semantic-comparison-command)
+3. [Command for visualisation of the found differences](#3-visualisation-of-the-found-differences)
+
+## 1. Snapshot generation commands
+
+These commands compile the compared project versions into LLVM IR and create
+so-called *snapshots* which contain the relevant LLVM IR files and additional
+metadata. There are several options for snapshot generation, choose the one
+which fits your needs. The chosen command should be run twice, once for each of
+the compared versions.
+
+### a) `build`: snapshot generation of `make`-based projects
+
+```sh
+diffkemp build PROJ_DIR SNAPSHOT_DIR [SYMBOL_LIST]
+```
+
+This is the default snapshot generation command for `make`-based projects. It
+takes the project located in `PROJ_DIR`, builds it into LLVM IR, and creates
+a snapshot for comparing the semantics of functions from `SYMBOL_LIST` (if no
+list is given, all exported functions from the project are considered).
+The snapshot is stored in `SNAPSHOT_DIR`. Warning: if `SNAPSHOT_DIR` exists,
+it will be overwritten.
+
+The command can be also used to generate a snapshot from a single C file.
+In this case, the path to the file should be given in place of `PROJ_DIR`.
+
+#### Options
+
+- `PROJ_DIR`: Path to a project's root directory or a path to a single C file.
+- `SNAPSHOT_DIR`: Output directory for storing the created snapshot.
+- `SYMBOL_LIST`: Path to file containing a list of symbols (each symbol on
+   a single line) which should be prepared for comparison.
+- `--no-opt-override`: Uses optimisation options provided in the project's
+  Makefile or specified with `--clang-append="-OX"`. With this option, DiffKemp
+  can potentially handle more complex refactoring (report fewer false
+  positives). However, this may reduce precision in identifying the exact
+  location (e.g. function or macro) of a semantic difference.
+- `--target TARGET`: Allows specifying make targets which should be used to
+  build the snapshot from the project.
+- `--reconfigure`: Reconfigures autotools-based project with `CC='DiffKemp
+  compiler wrapper'`.
```

Comment:
Slightly more understandable (at least to me) that it's not a string but a path to the wrapper.
```suggestion
- `--reconfigure`: Reconfigures autotools-based project with `CC=<DiffKemp
  compiler wrapper>`.
```